src/firecracker/src/main.rs

```diff
@@ -155,11 +155,17 @@ fn main_exitable() -> ExitCode {
         .arg(
             Argument::new(MMDS_CONTENT_ARG)
                 .takes_value(true)
+                // MMDS is not available when using `no-api`, so specifying a metadata
```

Comment:
Why would we want to do this? AFAIK using MMDS without the API thread is a valid use case: https://github.com/firecracker-microvm/firecracker/issues/1292

Reply:
This was a miss on my part - I was not aware that this PR has been merged. I removed the commit forbidding `--metadata` with `--no-api`.